misc_upload_config.md
1. create_user.py

```py
### content of "create_user.py" file
from account.models import UserBase, Buyer, Seller

buyer_names = ['arun_LyuK3C', 'alice_A52pLv', 'bob_PEmb5S'] 
buyer_passwords = ['FNNuYBmUU6PmhP2F', 'hyagZDy5qJQnSAhC', 'nyVF5brZt8YCBtw6'] 
buyer_emails = ['popika4897@ecofreon.com', 'sdizcxuduzrywxfqpx@sdvrecft.com', 'glgltawqsnrpttauhf@sdvrecft.com']

seller_names = ['eve_z636LZ', 'feynmann_5hgNeS', '1st_year_books']
seller_passwords = ['9k8ZvKyhArfLNBrr', 'bkTdL2drYZEsML7C', 'DHeNTF6r67CsSkC9']
seller_emails = ['padebo5687@epeva.com', 'hiyduhudra@vusra.com', 'lozoduvy@onekisspresave.com']

"""

email, user name, name, password

---
echo "from account.models import UserBase; UserBase.objects.create_superuser('admin_FdaB9B@admin.com', 'admin_FdaB9B', 'admin_FdaB9B', '4e=ETw:=cVB2]%c#')" | python manage.py shell

echo "from account.models import UserBase; UserBase.objects.create_superuser('admin_HUP2au@admin.com', 'admin_HUP2au', 'admin_HUP2au', '57?:4J4#+Ew9e]E#')" | python manage.py shell

echo "from account.models import UserBase; UserBase.objects.create_superuser('admin_y9sSM4@admin.com', 'admin_y9sSM4', 'admin_y9sSM4', '26TJbYjBh}#{F~<')" | python manage.py shell

echo "from account.models import UserBase; UserBase.objects.create_superuser('admin_yJG2M2@admin.com', 'admin_yJG2M2', 'admin_yJG2M2', '}q%Jx.TaD3Ha-P9')" | python manage.py shell

echo "from account.models import UserBase; UserBase.objects.create_superuser('admin_8ebK8x@admin.com', 'admin_8ebK8x', 'admin_8ebK8x', 'w5y;X>jdcFu<DDDu')" | python manage.py shell
---

---
echo "from account.models import UserBase; UserBase.objects.create_superuser('admin_kcz6My@admin.com', 'admin_kcz6My', 'admin_kcz6My', 'z78~gqu]D.u<g_')" | python manage.py shell

echo "from account.models import UserBase; UserBase.objects.create_superuser('admin_G8fv3H@admin.com', 'admin_G8fv3H', 'admin_G8fv3H', ',?4?wY!7_5aa$k')" | python manage.py shell

echo "from account.models import UserBase; UserBase.objects.create_superuser('admin_SZrd9q@admin.com', 'admin_SZrd9q', 'admin_SZrd9q', ':+\[2>LHLjd6xn_')" | python manage.py shell

echo "from account.models import UserBase; UserBase.objects.create_superuser('admin_9y3LqW@admin.com', 'admin_9y3LqW', 'admin_9y3LqW', 'R>V;2YUgyKz&wd')" | python manage.py shell

echo "from account.models import UserBase; UserBase.objects.create_superuser('admin_jqjU2E@admin.com', 'admin_jqjU2E', 'admin_jqjU2E', 'fruft}N7B&V#3$A')" | python manage.py shell
---



"""
```


2. server config

```
-------
sudo a2ensite django_project
sudo a2dissite 000-default.conf
-------
sudo tail /var/log/apache2/error.log
---------
git stash
git pull

edit settings.py file

python3 manage.py collectstatic

python3 manage.py check --deploy

SESSION_COOKIE_SECURE = True
CSRF_COOKIE_SECURE = True
-----

sudo chown :www-data fcs-online-marketplace/db.sqlite3
sudo chmod 664 fcs-online-marketplace/db.sqlite3
sudo chown :www-data fcs-online-marketplace/

sudo chown -R :www-data fcs-online-marketplace/media/
sudo chmod -R 775 fcs-online-marketplace/media

sudo chmod 775 fcs-online-marketplace/

sudo service apache2 restart
```

3. settings.py production

```py
"""
Django settings for core project.

Generated by 'django-admin startproject' using Django 3.2.8.

For more information on this file, see
https://docs.djangoproject.com/en/3.2/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/3.2/ref/settings/
"""
import os, json
from pathlib import Path
from django.contrib.messages import constants as messages

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent

# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/3.2/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
with open('/etc/config.json') as config_file:
    config = config_file.read()
    config = json.loads(config)

SECRET_KEY = config['SECRET_KEY']

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = False

ALLOWED_HOSTS = ['192.168.3.42', ]

# Application definition

INSTALLED_APPS = [
    # 2fa
    'django_otp',
    'django_otp.plugins.otp_static',
    'django_otp.plugins.otp_totp',
    'two_factor',
    # activity logging
    'django_db_logger',
    # internal
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    # user
    'store',
    'account',
    'orders',
    'seller',
    'crispy_forms',
    'django_social_share',
]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django_otp.middleware.OTPMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]

ROOT_URLCONF = 'core.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [BASE_DIR / 'templates'],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
                'store.context_processors.categories',
            ],
        },
    },
]

WSGI_APPLICATION = 'core.wsgi.application'

# Database
# https://docs.djangoproject.com/en/3.2/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': BASE_DIR / 'db.sqlite3',
    }
}

# Password validation
# https://docs.djangoproject.com/en/3.2/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
        'OPTIONS': {
            'min_length': 12,
        }
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]

# Internationalization
# https://docs.djangoproject.com/en/3.2/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'Asia/Kolkata'

USE_I18N = True

USE_L10N = True

USE_TZ = True

# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/3.2/howto/static-files/

STATIC_URL = '/static/'
STATIC_ROOT = os.path.join(BASE_DIR, 'assets')
STATICFILES_DIRS = [
    os.path.join(BASE_DIR, "static")
]

MEDIA_URL = '/media/'
MEDIA_ROOT = os.path.join(BASE_DIR, 'media/')
MAX_UPLOAD_SIZE = 5242880

# Stripe Payment
STRIPE_SECRET_KEY = config['STRIPE_SECRET_KEY']
STRIPE_PUBLIC_KEY = config['STRIPE_PUBLIC_KEY']
STRIPE_ENDPOINT_SECRET = config['STRIPE_ENDPOINT_SECRET']

# Custom user model
AUTH_USER_MODEL = 'account.UserBase'
LOGIN_REDIRECT_URL = '/account/dashboard'
LOGIN_URL = 'two_factor:login'

GOOGLE_RECAPTCHA_SECRET_KEY = config['GOOGLE_RECAPTCHA_SECRET_KEY']
CRISPY_TEMPLATE_PACK = 'bootstrap4'

# Default primary key field type
# https://docs.djangoproject.com/en/3.2/ref/settings/#default-auto-field


MESSAGE_TAGS = {
    messages.DEBUG: 'alert-secondary',
    messages.INFO: 'alert-info',
    messages.SUCCESS: 'alert-success',
    messages.WARNING: 'alert-warning',
    messages.ERROR: 'alert-danger',
}

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'

LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'formatters': {
        'verbose': {
            'format': '%(levelname)s %(asctime)s %(module)s %(process)d %(thread)d %(message)s'
        },
        'simple': {
            'format': '%(levelname)s %(asctime)s %(message)s'
        },
    },
    'handlers': {
        'db_log': {
            'level': 'DEBUG',
            'class': 'django_db_logger.db_log_handler.DatabaseLogHandler'
        },
    },
    'loggers': {
        'db': {
            'handlers': ['db_log'],
            'level': 'DEBUG'
        }
    }
}

SESSION_COOKIE_SECURE = False

AUTHENTICATION_BACKENDS = ('django.contrib.auth.backends.ModelBackend',)

EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
EMAIL_HOST = 'smtp.gmail.com'
EMAIL_PORT = 587
EMAIL_USE_TLS = True
EMAIL_HOST_USER = config['EMAIL_HOST_USER']
EMAIL_HOST_PASSWORD = config['EMAIL_HOST_PASSWORD']
```


core/settings.py
"""
Django settings for core project.

Generated by 'django-admin startproject' using Django 3.2.8.

For more information on this file, see
https://docs.djangoproject.com/en/3.2/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/3.2/ref/settings/
"""
import os
from pathlib import Path
from django.contrib.messages import constants as messages

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent

# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/3.2/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'django-insecure-auhlz&)r&f&ulrpfw1bwtm680uq$$irsia%5#+tb=^ozh@irlw'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = []

# Application definition

INSTALLED_APPS = [
    # 2fa
    'django_otp',
    'django_otp.plugins.otp_static',
    'django_otp.plugins.otp_totp',
    'two_factor',
    # activity logging
    'django_db_logger',
    # internal
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    # user
    'store',
    'account',
    'orders',
    'seller',
    'crispy_forms',
    'django_social_share',
]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django_otp.middleware.OTPMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]

ROOT_URLCONF = 'core.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [BASE_DIR / 'templates'],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
                'store.context_processors.categories',
            ],
        },
    },
]

WSGI_APPLICATION = 'core.wsgi.application'

# Database
# https://docs.djangoproject.com/en/3.2/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': BASE_DIR / 'db.sqlite3',
    }
}

# Password validation
# https://docs.djangoproject.com/en/3.2/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
        'OPTIONS': {
            'min_length': 12,
        }
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
    {
        'NAME': 'account.validators.UserNameSimilarityValidator',
    },
    {
        'NAME': 'account.validators.PasswordSpaceValidator',
    },
]

# Internationalization
# https://docs.djangoproject.com/en/3.2/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'Asia/Kolkata'

USE_I18N = True

USE_L10N = True

USE_TZ = True

# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/3.2/howto/static-files/

STATIC_URL = '/static/'
STATICFILES_DIRS = [
    os.path.join(BASE_DIR, "static")
]

MEDIA_URL = '/media/'
MEDIA_ROOT = os.path.join(BASE_DIR, 'media/')
MAX_UPLOAD_SIZE = 5242880

# Stripe Payment
STRIPE_SECRET_KEY = "sk_test_51JYjakSCFlACXPCD97CvOla0GDR9ePhorbwUoJcLn5K1UJjUxGbVVX0eMiO1A75cbmmBG3eZRiG8tizm7JYFtDzP00ygLLjKT3"
STRIPE_PUBLIC_KEY = "pk_test_51JYjakSCFlACXPCDlLJBNRRm1s9lFG71fMCJKnZ6K2LHcmZLvM6SHGVeWjxOebMaEMqIXSQEyBaigLGfevTgE4EF00MsaBdGUm"
STRIPE_ENDPOINT_SECRET = "whsec_khK16X9Am1CrAEO4C0oRMWu1oj7g9KPm"

# Custom user model
AUTH_USER_MODEL = 'account.UserBase'
LOGIN_REDIRECT_URL = '/account/dashboard'
LOGIN_URL = 'two_factor:login'

GOOGLE_RECAPTCHA_SECRET_KEY = '6LfrWOccAAAAAHqsiFhBDMsSC1nm2Ect6lPI-EKO'
CRISPY_TEMPLATE_PACK = 'bootstrap4'

# Default primary key field type
# https://docs.djangoproject.com/en/3.2/ref/settings/#default-auto-field


MESSAGE_TAGS = {
    messages.DEBUG: 'alert-secondary',
    messages.INFO: 'alert-info',
    messages.SUCCESS: 'alert-success',
    messages.WARNING: 'alert-warning',
    messages.ERROR: 'alert-danger',
}

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'

LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'formatters': {
        'verbose': {
            'format': '%(levelname)s %(asctime)s %(module)s %(process)d %(thread)d %(message)s'
        },
        'simple': {
            'format': '%(levelname)s %(asctime)s %(message)s'
        },
    },
    'handlers': {
        'db_log': {
            'level': 'DEBUG',
            'class': 'django_db_logger.db_log_handler.DatabaseLogHandler'
        },
    },
    'loggers': {
        'db': {
            'handlers': ['db_log'],
            'level': 'DEBUG'
        }
    }
}

SESSION_COOKIE_SECURE = False

AUTHENTICATION_BACKENDS = ('django.contrib.auth.backends.ModelBackend',)

EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
EMAIL_HOST = 'smtp.gmail.com'
EMAIL_PORT = 587
EMAIL_USE_TLS = True
EMAIL_HOST_USER = 'sriza20124@iiitd.ac.in'
EMAIL_HOST_PASSWORD = 'ohztiudeufnlczrn'
